### PR TITLE
update workspace list page

### PR DESCRIPTION
**Pull request by raintygao**
### Description

1. support search name and id
2. add multi select checkbox
3. support jump to edit workspace page
4. test suites will be included in integration PR

## Screenshot

image

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_list/index.tsx [hunk 1]
```diff
@@ -3,45 +3,50 @@
...
 import { useOpenSearchDashboards } from '../../../../../plugins/opensearch_dashboards_react/public';
-import { switchWorkspace } from '../utils/workspace';
+import { switchWorkspace, updateWorkspace } from '../utils/workspace';
+import { debounce } from '../utils/common';
+
+import { formatUrlWithWorkspaceId } from '../../utils';
+
+import { WORKSPACE_CREATE_APP_ID } from '../../../common/constants';
+
+const WORKSPACE_LIST_PAGE_DESCRIPTIOIN =
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_list/index.tsx:30]: Could you wrapper with i18n?

File: src/plugins/workspace/public/components/workspace_list/index.tsx [hunk 2]
```diff
@@ -3,45 +3,50 @@
...
-  const pageOfItems = useMemo(() => {
-    return workspaceList
-      .sort((a, b) => {
-        const compare = a[sortField].localeCompare(b[sortField]);
-        return sortDirection === 'asc' ? compare : -compare;
-      })
-      .slice(pageIndex * pageSize, (pageIndex + 1) * pageSize);
-  }, [workspaceList, pageIndex, pageSize, sortField, sortDirection]);
+  // Will be uesed when updating table actions
+  const [_, setSelection] = useState<WorkspaceAttribute[]>([]);
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_list/index.tsx:49]: Is that means this state won't be used? How about assign an empty function?

File: src/plugins/workspace/public/components/workspace_list/index.tsx [hunk 3]
```diff
@@ -52,6 +57,37 @@ export const WorkspaceList = () => {
...
+  const handleUpdateWorkspace = useCallback(
+    (id: string) => {
+      if (application && http) {
+        updateWorkspace({ application, http }, id);
+      }
+    },
+    [application, http]
+  );
+
+  const searchWorkspaces = useCallback((query: string, list: WorkspaceAttribute[]) => {
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_list/index.tsx:69]: Seems this searchWorkspaces only be used in below useMemo, how about move this function inside useMemo?

File: src/plugins/workspace/public/components/workspace_list/index.tsx [hunk 4]
```diff
@@ -79,46 +115,96 @@ export const WorkspaceList = () => {
...
   ];
 
-  const onTableChange = ({ page, sort }: CriteriaWithPagination<WorkspaceAttribute>) => {
-    const { field, direction } = sort!;
-    const { index, size } = page;
+  const workspaceCreateUrl = useMemo(() => {
+    if (!application || !http) {
+      return '';
+    }
+    return formatUrlWithWorkspaceId(
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_list/index.tsx:137]: How about use `cleanWorkspaceId` ?

Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_list/index.tsx:132]: Could you explain more?

File: src/plugins/workspace/public/components/workspace_list/index.tsx (hunk 2 above)
Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_list/index.tsx]: This state will be used when further mock up update.

File: src/plugins/workspace/public/components/workspace_list/index.tsx (hunk 4 above)
Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_list/index.tsx:132]: Format with an empty workspace id looks a little bit strange, is that means we just want to clear workspace id here?  We can call `cleanWorkspaceId` if only want to remove workspace id.

Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_list/index.tsx:132]: I don't find anything related when i search `cleanWorkspaceId` in repo.
Here I just want to generate a link to navigate to workspace create page, like [https://github.com/ruanyl/OpenSearch-Dashboards/blob/workspace/src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx#L121](url), do you have some ideas ?

Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_list/index.tsx:132]: The `cleanWorkspaceId` is in `src/core/utils/workspace.ts`, maybe you need to rebase first. The `applications.getUrlForApp` was already add the basePath, call `formatUrlWithWorkspaceId` will make code more complicated.

Comment (raintygao) [reply src/plugins/workspace/public/components/workspace_list/index.tsx]: Ok. I think we can make this link independent of workspace id.

**Review by wanglam — APPROVED** [review]

**Pull request merged**